### PR TITLE
pull-request

### PR DESCRIPTION
**Pull request by maku05**
The pull request is made to add a configurable maximum number of chars a user can enter in a tinymce editor. If the number is exceeded a error message will display above the corresponding tinymce editor. The message will display the character limit, the current char count and a button to automatically limit the chars to the maximum.

### CODE REVIEW

File: src/Resources/assets/js/contao-tinymce-bundle.js [hunk 1]
```diff
@@ -29,16 +33,142 @@ class TinyMceBundle {
...
+            let form = element.closest('form');
+            if(!form.classList.contains('tinymce-form')) {
+                form.classList.add('tinymce-form');
+            }
+        });
+
+    }
+
+    static initEventListeners() {
+        utilsBundle.event.addDynamicEventListener('click', '.tinymce-limitChars', (element) => {
```
Comment (Defcon0) [inline src/Resources/assets/js/contao-tinymce-bundle.js]: Please take care of codestyle: no camelcase in css classes, i.e. tinymce-limit-chars. Please check the rest of the code for this kind of issue.

File: src/Resources/assets/js/contao-tinymce-bundle.js [hunk 2]
```diff
@@ -7,8 +7,12 @@ import 'tinymce/themes/silver';
 class TinyMceBundle {
 
     static init() {
-        let tinyMceFields = document.querySelectorAll('textarea[data-tinymce="1"]'),
-            language = document.querySelector('html').getAttribute('lang');
+        TinyMceBundle.initTinymce();
+        TinyMceBundle.initEventListeners();
+    }
+
+    static initTinymce() {
```
Comment (Defcon0) [inline src/Resources/assets/js/contao-tinymce-bundle.js]: Please take care of a consistent writing of TinyMce -> initTiny**M**ce -> same for other methods like setTinymceError -> setTinyMceError

**Review by Defcon0 — CHANGES_REQUESTED** [review]
Only some code style things, rest looks good!

**Comment by Defcon0** [comment]
> Only some code style things, rest looks good!

One thing I saw in the frontend: Is it possible to make the alert disappear if the max chars limitation is met, i.e. the text is shorter than 10 again?

File: src/Resources/assets/js/contao-tinymce-bundle.js [hunk 3]
```diff
@@ -29,16 +33,142 @@ class TinyMceBundle {
...
+            TinyMceBundle.getTinyMceSetup(editor, config);
+        };
+
+        return config;
+    }
+
+
+    static getTinyMceSetup(editor, config) {
+        // set maxChars event listener
+        if('undefined' !== config.maxChars) {
```
Comment (AlexejKossmann) [inline src/Resources/assets/js/contao-tinymce-bundle.js]: This needs to be:  if( typeof 'undefined' !== config.maxChars)

Comment (maku05) [reply src/Resources/assets/js/contao-tinymce-bundle.js]: shouldn't it be if( 'undefined' !== typeof config.maxChars) ?

**Review by Defcon0 — APPROVED** [review]
Great. Thanks for implementing :-)

**Pull request merged**